packages/frontend/src/contexts/Web3Context.tsx

```diff
@@ -1,311 +1,361 @@
+            try {
+              await ethersProvider.send('eth_requestAccounts', [])
+            } catch (error) {
+              logger.error(error)
+            }
+          }
+          setProvider(ethersProvider)
+        } else {
+          setProvider(undefined)
         }
-      },
-      notify: {
-        enabled: false
-      },
-    })
-
-    return instance
-  }, [onboardNetworkId, isDarkMode])
-
-  async function handleWalletChange(wallet: any) {
-    try {
-      logger.debug('handleWalletChange')
-      // logger.debug('handleWalletChange wallet', wallet)
-
-      const _address = wallet?.accounts?.[0]?.address
-      if (_address) {
-        setAddress(Address.from(_address))
-      } else {
-        setAddress(undefined)
-      }
-
-      const connectedNetworkId = Number(wallet?.chains?.[0]?.id)
-      if (connectedNetworkId) {
-        setConnectedNetworkId(connectedNetworkId)
-      } else {
-        setConnectedNetworkId(undefined)
+      } catch(err) {
+        logger.error(err)
+        setProvider(undefined)
       }
+    }
+    update().catch(logger.error)
+  }, [account, connector])
 
-      if (wallet?.provider) {
-        const { icon, label, provider } = wallet
-        const ethersProvider = new ethers.providers.Web3Provider(provider, 'any')
-        if (provider.enable && !provider.isMetaMask) {
-          // needed for WalletConnect and some wallets
-          await provider.enable()
+  useEffect(() => {
+      try {
+        if (account) {
+          setAddress(Address.from(account))
         } else {
-          // note: this method may not be supported by all wallets
-          try {
-            await ethersProvider.send('eth_requestAccounts', [])
-          } catch (error) {
-            logger.error(error)
-          }
-        }
-        setProvider(ethersProvider)
-        setWalletName(label)
-
-        try {
-          const svg = new Blob([icon], { type: 'image/svg+xml' })
-          const url = URL.createObjectURL(svg)
-          setWalletIcon(url)
-        } catch (err: any) {
-          setWalletIcon('')
+          setAddress(undefined)
         }
-      } else {
-        setWalletName('')
-        setWalletIcon('')
-        setProvider(undefined)
+      } catch(err) {
+        logger.error(err)
         setAddress(undefined)
       }
-    } catch (err) {
-      logger.error(err)
-      setProvider(undefined)
-      setAddress(undefined)
+  }, [account])
+
+  useEffect(() => {
+    if (connector) {
+      setWalletName(getName(connector))
+      setWalletIcon(getIcon(connector))
+    } else {
+      setWalletName('')
+      setWalletIcon('')
     }
-  }
+  }, [connector])
 
-  function requestWallet() {
+  useEffect(() => {
     const update = async () => {
       try {
-        localStorage.clear()
-        const [primaryWallet] = onboard.state.get().wallets
-        if (primaryWallet) {
-          await onboard.disconnectWallet({ label: primaryWallet.label })
+        disconnectWallet()
```

Comment:
Should this be awaited? Little context about what these are doing, but looks like a possible edge-case race condition can set `connectorToUse.activate()` before completion of `connector.resetState()` in some weird case.

Not sure it even matters.